Fix - String Interpolation Leaks

cnoon opened the pull request:
This PR fixes memory leaks that were occurring in the string interpolation for the `description` and `debugDescription` implementations in the `Timeline` struct. None of these changes "should" be necessary, but unfortunately are at the moment. The main change was to switch over from string interpolation to string concatenation. I've been debugging with the sample project provided in #1232 and everything seems to be resolved with these changes.

jshier commented:
👍 

Though, perhaps make a `TODO` or otherwise indicate that the code should be moved back to interpolation when the leaks are fixed. Also, was there a bug filed with Apple for this issue? Radar or Swift project would work.

cnoon commented:
I agree @jshier. I put together the [Rdar-StringInterpolationMemoryLeak](https://github.com/Alamofire/Rdar-StringInterpolationMemoryLeak) project and filed [rdar://26761490](https://openradar.appspot.com/radar?id=5010235949318144) with Apple. I also added `NOTE`s to the code and added the rdar to the README in 6c1cbdd7.
